pkg/cloud/openstack/clients/machineservice.go

```diff
@@ -616,7 +626,7 @@ func (is *InstanceService) InstanceCreate(clusterName string, name string, clust
 	}
 
 	for _, portCreateOpts := range config.Ports {
-		port, err := getOrCreatePort(is, name+"-"+portCreateOpts.NameSuffix, portCreateOpts)
+		port, err := getOrCreatePort(is, name, portCreateOpts)
```

Comment:
```suggestion
		port, err := getOrCreatePort(is, name+"-"+portCreateOpts.NameSuffix, portCreateOpts)
```
you should leave this, this is for ports created  with the port directive, and it needs the name suffix to ensure the name is unique.